### ISSUE TITLE
singlecopy filter doesn't seem to work as expected

### ISSUE THREAD

**Issue by glennhickey** [ComparativeGenomicsToolkit/cactus#314]
With `singleCopy` on, I hope to prevent blocks containing the same species more than once.  But this doesn't happen.  The [filter function](https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/caf/impl/filtering.c#L157-L160) itself seems pretty foolproof, but maybe it's not getting called for every pinch...

Here's a maf block from a very small test
```
(simCow_chr6:1.0,simDog_chr6:1.0)Anc0;                                                                                                                                                                                              
simCow_chr6 starTest/simCow.chr6                                                                                                                                                                                                    
simDog_chr6 starTest/simDog.chr6 
```
and running `cactus` with the `alignmentFilter="singleCopy"` in `caf`, and even turning off `bar` gives an output with this MAF block:

```
a                                                                                                                                                                                                                                   
s  Anc0.Anc0refChr6  51351 57 +  95926 CAAACTAACAGGGTAAGATGCATTTGTATACTCCGGACCTGGGAGCCAATAGCCTGG                                                                                                                                    
s  simCow_chr6.simCow.chr6 42306 57 +  602619   TGAACTAACAGGGTAAGAAGGATTCGTATACACCGGACCTGGGAGCCAATAGCCTAG                                                                                                                           
s  simCow_chr6.simCow.chr6 467018   57 +  602619   CAAACTAACAGGGTAAGATGCATTTGGATACTCTAGACCTAGGGGTGAATAGGCCGG                                                                                                                        
s  simDog_chr6.simDog.chr6 37132 57 +  593897   CAAACCAACAGACTGAGATGCATTCCTATAGTCCAGATCTGGGAGCCAAGAACCTGG                                                                                                                           
s  simDog_chr6.simDog.chr6 456991   57 +  593897   TGAACTAACAGGGTAAGATGCATTTATATACTCCGGACCTTGGAGGCAGTAACCTGG                                                                                                                        
```

**Comment by glennhickey** [comment]
Oh I see, when I specify `singleCopy` it applies only as a primary alignment filter and the secondaries are still free to pinch duplications.  I guess it should apply to both by default, or perhaps it's better to have two parameters: one for a primary and one for a secondary filter that we can mix and match.

**Comment by glennhickey** [comment]
This was resolved in #315 which changed the single copy filter (and all other formerly primary-only filters) to operate on both primary and secondary alignments
